src/main/java/reiff/finalproject/ImageFrame.java

```diff
@@ -0,0 +1,33 @@
+package reiff.finalproject;
+
+import javax.swing.*;
+import java.awt.*;
+import java.net.URL;
+
+public class ImageFrame extends JFrame {
+    public ImageFrame(String imageUrl) {
+        setTitle("Art Image");
+        setSize(800, 600);
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+        setLayout(new BorderLayout());
+
+        try {
+            ImageIcon imageIcon = new ImageIcon(new URL(imageUrl));
+            Image image = imageIcon.getImage();
+            int width = image.getWidth(null);
```

Comment:
You don't need width anymore right?

In fact now you can just download the image the way I instructed you to so you don't need two ImageIcons the way you have it now.